packages/babel-plugin-transform-goober/index.js

```diff
@@ -1,9 +1,86 @@
-module.exports = function({ types: t }, options = {}) {
+const nodePath = require('path');
+
+/**
+ * Prepend #__PURE__ comment to help minifiers with
+ * dead code elminiation (=DCE)
+ */
+function prependPureComment(node) {
+    const comments = node.leadingComments || (node.leadingComments = []);
+    comments.push({
+        type: 'CommentBlock',
+        value: '#__PURE__'
+    });
+}
+
+function createAssignment(t, left, right) {
+    return t.expressionStatement(t.assignmentExpression('=', left, right));
+}
+
+module.exports = function ({ types: t }, options = {}) {
     const name = options.name || 'styled';
+    // Enable pure by default if it is not set by the user
+    const pure = !('pure' in options) ? true : options.pure;
 
     return {
         name: 'transform-goober',
         visitor: {
+            TaggedTemplateExpression(path, state) {
+                if (t.isIdentifier(path.node.tag) && path.node.tag.name === 'css') {
+                    if (pure) {
+                        prependPureComment(path.node);
+                    }
+                } else if (
+                    t.isIdentifier(path.node.tag.callee) &&
+                    path.node.tag.callee.name === name
+                ) {
+                    if (pure) {
+                        prependPureComment(path.node);
+                    }
+
+                    if (options.displayName) {
+                        const variable = path.findParent((path) => path.isVariableDeclaration());
+                        if (variable && variable.node.declarations.length === 1) {
+                            const decl = variable.node.declarations[0];
+
+                            if (t.isIdentifier(decl.id)) {
+                                // Add displayName to goober components for easier debugging
+                                const displayName = variable.insertAfter(
+                                    createAssignment(
+                                        t,
+                                        t.MemberExpression(decl.id, t.identifier('displayName')),
```

Comment:
Question, is it possible to change `displayName` into something else? Right now `Styled.displayName` adds like `+7B` which are coming from `display` I think. So my thinking is that could we change this to `name` and still be picked-up by devtools and such?

Reply:
Actually, nope. It's not the `displayName` :D I'll be back. Chasing the `+7B` addition.

Reply:
Another key won't work I'm afraid :S Setting the function name is a noop in most browsers and will throw in older ones. That's why we use `displayName` in `preact/compat` and it's the same property the react ecosystem uses for the same reasons.